### PR TITLE
xcube edit metadata tool

### PR DESCRIPTION
**Pull request by AliceBalfanz**
This PR adds a tool called `xcube edit` , which allows the user to edit the metadata of an existing xcube datataset with the help of a yml file. The PR is related to issue #170.

### CODE REVIEW

**Comment by AliceBalfanz** [comment]
Closing this PR, and will reopen after implementing the ideas of Wednesdays discussion with @forman  and @dzelge .

File: CHANGES.md [hunk 1]
```diff
@@ -2,8 +2,10 @@
 
 ### Enhancements
 
-- `xcube serve` now recognises xcube datasets with consolidated 
-  metadata. (#141) 
+- Added new CLI tool `xcube edit` and API function `xcube.api.edit_metadata`
+  which allows to edit the metadata of an existing xcube dataset. (#170)
```
Comment (forman) [inline CHANGES.md]:
```suggestion
  which allows editing the metadata of an existing xcube dataset. (#170)
```

File: CHANGES.md [hunk 2]
```diff
@@ -2,8 +2,10 @@
...
+- `xcube serve` now recognises xcube datasets with consolidated
+  metadata. (#141)
```
Comment (forman) [inline CHANGES.md]:
```suggestion
  metadata consolidated by the `xcube opmimize` command. (#141)
```

File: CHANGES.md [hunk 3]
```diff
@@ -2,8 +2,10 @@
 
 ### Enhancements
 
-- `xcube serve` now recognises xcube datasets with consolidated 
-  metadata. (#141) 
+- Added new CLI tool `xcube edit` and API function `xcube.api.edit_metadata`
+  which allows to edit the metadata of an existing xcube dataset. (#170)
+- `xcube serve` now recognises xcube datasets with consolidated
```
Comment (forman) [inline CHANGES.md]:
```suggestion
- `xcube serve` now recognises xcube datasets with
```

File: docs/source/cli/xcube_edit.rst [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      -I, --in-place           Edit the cube in place. Ignores output path.
+      --help                   Show this message and exit.
+
+
+
+
+Examples
+========
+
+The global attributes of the demo xcube dataset  `demo cube-1-250-250.zarr`_ do not contain the creators name
```
Comment (forman) [inline docs/source/cli/xcube_edit.rst]: Say, where will users find that one.

File: xcube/util/edit.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    :param in_place: Whether to modify the dataset in place.
+           If False, a copy is made and *output_path* must be given.
+    :param monitor: A progress monitor.
+    :param exception_type: Type of exception to be used on value errors.
+    """
+
+    if not os.path.isfile(os.path.join(input_path, '.zgroup')):
+        raise exception_type('Input path must point to ZARR dataset directory.')
+
+    input_path = os.path.abspath(os.path.normpath(input_path))
```
Comment (forman) [inline xcube/util/edit.py]: this line should be *before* the upper one, because you change `input_path`

**Pull request merged**